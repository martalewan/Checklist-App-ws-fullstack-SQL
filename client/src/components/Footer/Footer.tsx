const Footer = () => (
    <footer className="footer-wrapper">
        <p className="footer-wrapper__link">Made by <a href="https://martalewan.com/">Marta Lewan</a></p>
        <p className="footer-wrapper__link">2022</p>
    </footer>
);

export default Footer;