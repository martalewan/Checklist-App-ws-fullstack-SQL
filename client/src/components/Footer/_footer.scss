.footer-wrapper {
  position: absolute;
  left: 0;
  bottom: 0;
  height: 50px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: var(--white);
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.095);

  &__link {
    margin: 0 2rem;
    margin-left: 2%;
    margin-right: 2%;
    font-size: 12px;
    color: var(--dark-grey);
  }
}

.footer-wrapper__link a {
  color: var(--blue);
}