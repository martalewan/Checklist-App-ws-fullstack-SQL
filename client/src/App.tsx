import TodoListPage from './pages/TodoListPage';
import { useNavigate } from "react-router-dom";
import HomePage from './pages/HomePage';
import { Route, Routes } from "react-router-dom";
import { createTodoList } from './services/Api';
import Navigation from './components/Navigation/Navigation';
import Footer from './components/Footer/Footer';
import ListsPage from './pages/ListsPage';

type CreateNewTodoList = (data: Object) => any;

const App = () => {
  const navigate = useNavigate();

  const createNewTodoList: CreateNewTodoList = (data : Object) => {
    createTodoList(data).then((res: { id: string; }) => {
      navigate('/todo-list/' + res.id);
    }).catch((err: any) => {
        console.error(err);
    })
  }

  return (
    <>
      <Navigation createNewTodoList={createNewTodoList}/>
      <div className="App">
        <Routes>
          <Route path="/todo-list/:id" element={<TodoListPage /> } />
          <Route path="/" element={<HomePage />} />
          <Route path="/lists" element={<ListsPage />} />
        </Routes>
      </div>
      <Footer />

    </>
  );
}

export default App;
