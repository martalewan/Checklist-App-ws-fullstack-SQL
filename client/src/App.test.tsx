
import { render } from '@testing-library/react';
import App from './App';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from "react-router-dom";
import Footer from './components/Footer/Footer';

test('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(
    <Router>
      <App />
    </Router>, div
  );
  ReactDOM.unmountComponentAtNode(div);
});

test('Snapshot', () => {
  const { asFragment } = render(<Footer />)
  expect(asFragment()).toMatchSnapshot();
});