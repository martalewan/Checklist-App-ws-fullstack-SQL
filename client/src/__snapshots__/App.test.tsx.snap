// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer-wrapper"
  >
    <p
      class="footer-wrapper__link"
    >
      Made by 
      <a
        href="https://martalewan.com/"
      >
        Marta Lewan
      </a>
    </p>
    <p
      class="footer-wrapper__link"
    >
      2022
    </p>
  </footer>
</DocumentFragment>
`;
